Refactor files about identity types and homotopies (#1014)
UniMath/agda-unimath: src/foundation/whiskering-homotopies-concatenation.lagda.md

# Whiskering homotopies with respect to concatenation

```agda
module foundation.whiskering-homotopies-concatenation where
```

<details><summary>Imports</summary>

```agda
open import foundation.functoriality-dependent-function-types
open import foundation.universe-levels
open import foundation.whiskering-identifications-concatenation
open import foundation.whiskering-operations

open import foundation-core.equivalences
open import foundation-core.homotopies
```

</details>

## Idea

Consider a homotopy `H : f ~ g` and a homotopy `K : I ~ J` between two
homotopies `I J : g ~ f`. The
{{#concept "left whiskering" Disambiguation="homotopies with respect to concatenation" Agda=left-whisker-concat-htpy}}

> of `H` and `K` is a homotopy `H ∙ I ~ H ∙ J`. In other words, left whiskering of

Reviewer: ```suggestion
of `H` and `K` is a homotopy `H ∙h I ~ H ∙h J`. In other words, left whiskering of
```

homotopies with respect to concatenation is a
[whiskering operation](foundation.whiskering-operations.md)

```text
  (H : f ~ g) {I J : g ~ h} → I ~ J → H ∙h I ~ H ∙h K.
```

Similarly, we introduce

> {{#concept 'right whiskering" Disambiguation="homotopies with respect to concatenation' Agda=right-whisker-concat-htpy}}

Reviewer: ```suggestion
{{#concept r"ight whiskering" Disambiguation="homotopies with respect to concatenation" Agda=right-whisker-concat-htpy}}
```

to be an operation

```text
  {H I : f ~ g} → H ~ I → (J : g ~ h) → H ∙h J ~ I ∙h J.
```

## Definitions

### Left whiskering of homotopies with respect to concatenation

Left whiskering of homotopies with respect to concatenation is an operation

```text
  (H : f ~ g) {I J : g ~ h} → I ~ J → H ∙h I ~ H ∙h K.
```

We implement the left whiskering operation of homotopies with respect to
concatenation as an instance of a general left whiskering operation.

```agda
module _
  {l1 l2 : Level} {A : UU l1} {B : A → UU l2}
  where

  left-whisker-concat-htpy :

>     left-whiskering-operation ((x : A) → B x) _~_ _∙h_ _~_

Reviewer: ```suggestion
    left-whiskering-operation ((x : A) → B x) (_~_) (_∙h_) (_~_)
```

  left-whisker-concat-htpy H K x = left-whisker-concat (H x) (K x)
```

### Right whiskering of homotopies with respect to concatenation

Right whiskering of homotopies with respect to concatenation is an operation

```text
  {H I : f ~ g} → H ~ I → (J : g ~ h) → H ∙h J ~ I ∙h J.
```

We implement the right whiskering operation of homotopies with respect to
concatenation as an instance of a general right whiskering operation.

```agda
module _
  {l1 l2 : Level} {A : UU l1} {B : A → UU l2}
  where

  right-whisker-concat-htpy :

>     right-whiskering-operation ((x : A) → B x) _~_ _∙h_ _~_

Reviewer: ```suggestion
    right-whiskering-operation ((x : A) → B x) (_~_) (_∙h_) (_~_)
```
